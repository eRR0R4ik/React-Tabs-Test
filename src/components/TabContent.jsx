import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Description from './Description.jsx';
import Details from './Details.jsx';


class TabContent extends Component {
  render() {
    let activeClass = this.props.activeId;

    /**
     * render TabContent for each element of data
     */
    let content = this.props.data.map((item, index) => {
      return (
        <div
          key={`content ${item.id}`}
          className={`phases-tabbox ${activeClass === index ? '__active' : ''}`}
        >
          <div className="phases-content">
            <Description
              picture={item.picture}
              name={item.name}
              description={item.description}
            />
            <hr />
            <Details />
          </div>
        </div>
      )
    });

    return (
      <div className="phases-container">{content}</div>
    );
  }
}

TabContent.propTypes = {
  activeId: PropTypes.number.isRequired,
  data: PropTypes.array.isRequired,
};

export default TabContent;
