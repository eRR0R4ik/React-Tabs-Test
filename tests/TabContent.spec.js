import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import TabContent from '../src/components/TabContent.jsx';

describe('Component: TabContent', () => {
  const minProps = {
    data: [
      {
        picture: 'mockPics1',
        name: 'mockName1',
        description: 'mockDescription1'
      },
      {
        picture: 'mockPics2',
        name: 'mockName2',
        description: 'mockDescription2'
      },
    ],
    activeId: 1
  }

  it('should render <TabContent /> components', () => {
    expect(shallow(<TabContent {...minProps} />).length).toEqual(1);
  });

  it('should render correctly', () => {
    const component = shallow(
      <TabContent {...minProps} />
    );
    expect(shallowToJson(component)).toMatchSnapshot();
  });

  it('renders an `.__active`', () => {
    const component = shallow(<TabContent {...minProps} />);
    expect(component.find('.__active').length).toEqual(1);
  });
});
