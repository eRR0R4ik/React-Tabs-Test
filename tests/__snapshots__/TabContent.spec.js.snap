// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: TabContent should render correctly 1`] = `
<div
  className="phases-container"
>
  <div
    className="phases-tabbox "
    key="content undefined"
  >
    <div
      className="phases-content"
    >
      <Description
        description="mockDescription1"
        name="mockName1"
        picture="mockPics1"
      />
      <hr />
      <Details />
    </div>
  </div>
  <div
    className="phases-tabbox __active"
    key="content undefined"
  >
    <div
      className="phases-content"
    >
      <Description
        description="mockDescription2"
        name="mockName2"
        picture="mockPics2"
      />
      <hr />
      <Details />
    </div>
  </div>
</div>
`;
